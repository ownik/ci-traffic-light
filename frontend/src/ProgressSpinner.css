.lds-roller,
.lds-roller div,
.lds-roller div:after {
  box-sizing: border-box;
}
.lds-roller {
  display: inline-block;
  position: relative;
  width: 40px;
  height: 40px;
}
.lds-roller div {
  animation: lds-roller 2s cubic-bezier(0.5, 0, 0.5, 1) infinite;
  transform-origin: 20px 20px;
}
.lds-roller div:after {
  content: ' ';
  display: block;
  position: absolute;
  width: 3.6px;
  height: 3.6px;
  border-radius: 50%;
  background: #fff;
  margin: -1.8px 0 0 -1.8px;
}
.lds-roller div:nth-child(1) {
  animation-delay: -0.06s;
}
.lds-roller div:nth-child(1):after {
  top: 31.31371px;
  left: 31.31371px;
}
.lds-roller div:nth-child(2) {
  animation-delay: -0.12s;
}
.lds-roller div:nth-child(2):after {
  top: 33.85641px;
  left: 28px;
}
.lds-roller div:nth-child(3) {
  animation-delay: -0.18s;
}
.lds-roller div:nth-child(3):after {
  top: 35.45481px;
  left: 24.1411px;
}
.lds-roller div:nth-child(4) {
  animation-delay: -0.24s;
}
.lds-roller div:nth-child(4):after {
  top: 36px;
  left: 20px;
}
.lds-roller div:nth-child(5) {
  animation-delay: -0.3s;
}
.lds-roller div:nth-child(5):after {
  top: 35.45481px;
  left: 15.8589px;
}
.lds-roller div:nth-child(6) {
  animation-delay: -0.36s;
}
.lds-roller div:nth-child(6):after {
  top: 33.85641px;
  left: 12px;
}
.lds-roller div:nth-child(7) {
  animation-delay: -0.42s;
}
.lds-roller div:nth-child(7):after {
  top: 31.31371px;
  left: 8.68629px;
}
.lds-roller div:nth-child(8) {
  animation-delay: -0.48s;
}
.lds-roller div:nth-child(8):after {
  top: 28px;
  left: 6.14359px;
}
@keyframes lds-roller {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}
