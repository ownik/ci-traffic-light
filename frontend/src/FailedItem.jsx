import React from 'react';
import PropTypes from 'prop-types';
import ProgressSpinner from './ProgressSpinner';

const FailedItem = ({ displayName, href, running }) => (
  <li>
    <span>
      {!href || href === '' ? displayName : <a href={href}>{displayName}</a>}
    </span>
    {running && <ProgressSpinner />}
  </li>
);

FailedItem.propTypes = {
  displayName: PropTypes.string,
  href: PropTypes.string,
  running: PropTypes.bool,
};

FailedItem.defaultProps = {
  href: '',
};

export default FailedItem;
