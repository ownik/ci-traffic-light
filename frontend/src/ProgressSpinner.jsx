import React from 'react';
import PropTypes from 'prop-types';

import './ProgressSpinner.css';

const ProgressSpinner = () => (
  <div className="lds-roller">
    <div></div>
    <div></div>
    <div></div>
    <div></div>
    <div></div>
    <div></div>
    <div></div>
    <div></div>
  </div>
);

ProgressSpinner.propTypes = {
  visible: PropTypes.bool,
};

ProgressSpinner.defaultProps = {
  visible: false,
};

export default ProgressSpinner;
