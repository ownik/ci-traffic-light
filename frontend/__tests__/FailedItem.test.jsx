import React from 'react';
import { shallow } from 'enzyme';
import FailedItem from '../src/FailedItem';
import ProgressSpinner from '../src/ProgressSpinner';

describe('FailedItem', () => {
  test('renders without crashing', () => {
    shallow(<FailedItem />);
  });

  test('display name render as text', () => {
    expect(
      shallow(<FailedItem displayName="Some display test" />).html()
    ).toMatchSnapshot();
  });

  test('display name with href render as link', () => {
    expect(
      shallow(
        <FailedItem
          displayName="Some display test"
          href="http://localhost/href"
        />
      ).html()
    ).toMatchSnapshot();
  });

  test('no running - no progress spinner', () => {
    const wrapper = shallow(<FailedItem />);
    expect(wrapper.find(ProgressSpinner)).toHaveLength(0);
  });

  test('running - have one progress spinner', () => {
    const wrapper = shallow(<FailedItem running={true} />);
    expect(wrapper.find(ProgressSpinner)).toHaveLength(1);
  });
});
