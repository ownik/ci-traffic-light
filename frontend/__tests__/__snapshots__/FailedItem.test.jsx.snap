// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FailedItem display name render as text 1`] = `"<li>Some display test</li>"`;

exports[`FailedItem display name with href render as link 1`] = `"<li><a href=\\"http://localhost/href\\">Some display test</a></li>"`;
